pkg/chunk/azure/blob_storage_client.go

```diff
@@ -110,13 +111,26 @@ func NewBlobStorage(cfg *BlobStorageConfig) (*BlobStorage, error) {
 func (b *BlobStorage) Stop() {}
 
 func (b *BlobStorage) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, error) {
+	var cancel context.CancelFunc
 	if b.cfg.RequestTimeout > 0 {
-		// The context will be cancelled with the timeout or when the parent context is cancelled, whichever occurs first.
-		var cancel context.CancelFunc
 		ctx, cancel = context.WithTimeout(ctx, b.cfg.RequestTimeout)
-		defer cancel()
 	}
 
+	rc, err := b.getObject(ctx, objectKey)
+	if cancel != nil {
```

Comment:
if `cancel` is nil, `err` is ignored

Reply:
An easier way to implement it may be assigning a no-op static function to `cancel` if timeout is not not set, so that you don't have to worry whether `cancel` is defined or not (and performance wise there's no real penalty doing so).

Reply:
@sandeepsukhani Have you got a chance to look at this comment?

Reply:
Sorry, I was held up with some other work. Will take care of this tomorrow.

Reply:
I have pushed this change. Thanks for the review!